### PR TITLE
Tidy up test names and descriptions

### LINKED ISSUE #274: Tidy up test naming
Our current Actions setup includes a lot of redundancy in the test naming. We should tidy these up and generally check test organization so things are easier to pick out visually in the PR check list.

### PR DESCRIPTION
**Pull request by mbestavros**
Hopefully, this makes things a little more organized.

Resolves #274.

### CODE REVIEW

File: .github/workflows/miscellaneous.yml [hunk 1]
```diff
@@ -1,35 +1,31 @@
-name: Enarx Standard Tests
+name: misc
 # Automated testing run on all commits.
```
Comment (npmccallum) [inline .github/workflows/miscellaneous.yml]: Remove this comment.

File: .github/workflows/cargo.yml [hunk 2]
```diff
@@ -1,11 +1,11 @@
-name: Enarx Rust Tests
+name: cargo
 # Automated testing specific to Enarx's Rust code.
```
Comment (npmccallum) [inline .github/workflows/cargo.yml]: Remove this comment.

File: .github/workflows/miscellaneous.yml [hunk 3]
```diff
@@ -1,35 +1,31 @@
...
-    container: quay.io/enarx/fedora-test
-    steps:
-    - uses: actions/checkout@v2
-  
-    # Ensure all code-related files have a newline at end of file.
-    - name: Cargo edition
-      run: .tests/eof-newline
+    # Ensure all .rs files have an Apache 2.0 SPDX license ID.
+    - name: All .rs files have SPDX IDs
+      run: .tests/rs-spdx-license
```
Comment (npmccallum) [inline .github/workflows/miscellaneous.yml]: Rename all test scripts to match job name.

File: .github/workflows/miscellaneous.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    container: quay.io/enarx/fedora-test
+    steps:
+    - uses: actions/checkout@v2
+  
+    # Ensure all code-related files have a newline at end of file.
+    - name: All code files have a newline at EOF
+      run: .tests/misc-eof
+
+  # Enforce correct license files and headers.
+  correct-licenses:
```
Comment (npmccallum) [inline .github/workflows/miscellaneous.yml]:
```suggestion
  misc-licenses:
```

Also, update test file names.

Comment (mbestavros) [reply .github/workflows/miscellaneous.yml]: Done.

**Pull request merged**